book-service-rest/src/main/java/com/example/Library/controller/UsersController.java

```diff
@@ -0,0 +1,106 @@
+package com.example.Library.controller;
+
+
+import com.example.Library.dto.UserDTO;
+import com.example.Library.exceptions.IncorrectGenderException;
+import com.example.Library.exceptions.IncorrectRoleException;
+import com.example.Library.exceptions.IncorrectUserStatusException;
+import com.example.Library.exceptions.UserAlreadyExists;
+import com.example.Library.repositories.UserRepository;
+import com.example.Library.services.UserService;
+import com.example.Library.services.impl.LoginAttemptService;
+import io.swagger.v3.oas.annotations.Operation;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+
+@RestController
+@RequestMapping("/api/v1/users")
+@Slf4j
+public class UsersController {
+
+    private final UserService userService;
+    private final UserRepository userRepository;
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+    @Autowired
+    private LoginAttemptService loginAttemptService;
+
+    @Autowired
+    public UsersController(UserService userService, UserRepository userRepository) {
+        this.userService = userService;
+        this.userRepository = userRepository;
+    }
+   // @PreAuthorize("hasAuthority('books:create')")
+    @GetMapping()
+    @Operation(summary = "Get all blocked users")
+    public ArrayList<String> getAllBlockedUsers(){
+        return loginAttemptService.getAllBlockedUsers();
+    }
+
+    @PreAuthorize("hasAuthority('books:create')")
+    @PostMapping()
+    @Operation(summary = "Add a user to the library using parameters.")
+    public UserDTO addUser(@RequestParam(value = "firstName") String firstName,
```

Comment:
wouldn't it be easier to accept UserDTO as an input, instead of many separate parameters? And then do a validation of a DTO